node/src/components/linear_chain_sync.rs

```diff
@@ -312,7 +495,48 @@ where
     REv: ReactorEventT<I>,
 {
     effect_builder.fetch_block(block_hash, peer).option(
-        move |value| Event::GetBlockResult(block_hash, Some(value)),
-        move || Event::GetBlockResult(block_hash, None),
+        move |value| Event::GetBlockHashResult(block_hash, Some(value)),
+        move || Event::GetBlockHashResult(block_hash, None),
     )
 }
+
+fn fetch_block_at_height<I: Send + Copy + 'static, REv>(
+    effect_builder: EffectBuilder<REv>,
+    peer: I,
+    block_height: u64,
+) -> Effects<Event<I>>
+where
+    REv: ReactorEventT<I>,
+{
+    effect_builder
+        .fetch_block_by_height(block_height, peer)
+        .option(
+            move |fetch_result| match fetch_result {
+                FetchResult::FromPeer(result, _) => match *result {
+                    BlockByHeight::Absent(ret_height) => {
+                        assert!(
```

Comment:
`assert_eq!`?

Reply:
Sure, but on the second thought I don't think I should panic here…

Reply:
Done.